corehq/messaging/scheduling/async_handlers.py

```diff
@@ -84,23 +84,12 @@ def schedule_user_organization_recipients_response(self):
         return self._get_user_organization_response()
 
     def _get_user_organization_response(self, case_sharing_only=False):
-        domain = self.request.domain
-        query = self.data.get('searchString')
-        result = (
-            SQLLocation
-            .active_objects
-            .filter(domain=domain, name__icontains=query)
-            .order_by('name')
-            .values_list('location_id', 'name')
-        )
-
-        if case_sharing_only:
-            result = result.filter(location_type__shares_cases=True)
-
-        return [
-            {'id': row[0], 'text': row[1]}
-            for row in result[:10]
-        ]
+        from corehq.apps.locations.views import LocationOptionsController
+        controller = LocationOptionsController(self.request, self.request.domain,
+                                               self.data.get('searchString'),
+                                               case_sharing_only=case_sharing_only)
+        (count, results) = controller.get_options()
+        return results[:10]
```

Comment:
will this support pagination?

Reply:
No, `select2_handler.js` doesn't support pagination.

Reply:
So this is a step towards fully solving this eventually?

Reply:
I think ideally we'll eventually get rid of `select2_handler.js` just because we have too many ways to make select2s, but this PR doesn't really get us any closer to that goal.